content/guides/getstarted/rust-to-solana.md

```diff
@@ -0,0 +1,273 @@
+setup details, restrictions, macro changes, and compute limits. Additionally,
+this guide will cover the development environments and frameworks needed to
+start with Solana.
+
+By the end of this guide, Rust developers will understand the differences they
+need to know to start their Solana journeys.
+
+## Understanding the Core Differences
+
+First, note that this guide aims at understanding the differences in using Rust
+as a language when working with Solana. It won’t cover
+[Blockchain or Solana basics](https://solana.com/learn/blockchain-basics).
+
+It also won’t cover core Solana concepts that must be understood in order to
+program in Solana, such as:
+
+- [Programs](https://solana.com/docs/core/programs) - Solana’s version of smart
+  contracts
+- [Accounts](https://solana.com/docs/core/accounts) - A record in the Solana
+  ledger that either holds data (a data account) or is an executable program
+- Various [fees](https://solana.com/docs/core/fees#why-pay-transaction-fees) -
+  Such as base fee, priority fee, and rent
+- [Transactions](https://solana.com/docs/core/transactions) - Interactions with
+  the network that contain instructions, signatures, and more.
+
+For more information on those core concepts, check out the
+[Solana developer documentation](https://solana.com/developers).
+
+Let’s now look at the differences in **project setup**.
+
+## Key Setup Details
+
+Onchain programs for Solana in Rust are still Rust programs at heart. They
+still follow the standard Rust project with a `/src` folder and `Cargo.toml`
+file in the root. That said, there are several key differences.
+
+### Project Dependencies
+
+To get started, the
+[solana-program crate](https://crates.io/crates/solana-program) is required for
+every onchain Solana program written with Rust. This is the base library for
+all onchain Rust programs. The library defines macros for the required
+**program entrypoint** _(see below)_, **core data types**, **logging macros**,
+and more.
+
+### Program Entrypoint
+
+Instead of a `main` function, Solana programs use the `entrypoint!` macro. This
+symbol is exported and then called by the Solana runtime when the program runs.
+The entrypoint macro calls a given function, which must have the following type
+signature:
+
+```rust
+pub fn process_instruction( program_id: &Pubkey, accounts: &[AccountInfo],
+instruction_data: &[u8], ) -> ProgramResult {
+
+        //program code goes here
+
+    }
+```
+
+These three parameters are passed to every onchain program.
+
+1. The `program_id` is the public key of the current program.
+2. The `accounts` are all accounts that are required to process the instruction.
+3. The `instruction_data` is data specific to that instruction.
+
+Every program must then call the `entrypoint!` macro on the instruction, as
+follows:
+
+```rust
+entrypoint!(process_instruction);
+```
+
+### Building and Testing
+
+After installing the Solana command-line tools, projects can be built to target
```

Comment:
Let's add a link to the install guide:
PS: Prettier will need to be run on this

```suggestion
After [installing the Solana command-line tools](https://solana.com/developers/guides/getstarted/setup-local-development#3-install-the-solana-cli), projects can be built to target
```